compiler/AST/ForallStmt.cpp

```diff
@@ -389,16 +455,24 @@ BlockStmt* ForallStmt::build(Expr* indices, Expr* iterator,
     indices = new UnresolvedSymExpr("chpl__elidedIdx");
   checkIndices(indices);
 
-  if (!forall_intents)
-    forall_intents = new ForallIntents();
-
-  ForallStmt* fs = new ForallStmt(zippered, body, forall_intents);
+  ForallStmt* fs = new ForallStmt(zippered, body);
+
+  if (forall_intents) {
+    // transfer from 'forall_intents' into 'fs'
+    // todo: how to make the parser produce AList directly?
+    AList& fis = fs->forallIntents();
+    for (int i = 0; i < forall_intents->numVars(); i++)
+      fis.insertAtTail(new ForallIntent(forall_intents->fIntents[i],
+                                        forall_intents->fiVars[i],
+                                        forall_intents->riSpecs[i]));
+    delete forall_intents;
```

Comment:
We can't get rid of ForallIntents class right now. It is still used for representing forall loops via BlockStmt. Eliminating ForallIntents is part of the "convert the remaining forall loops" item in #6797.